packages/docusaurus/lib/client/docusaurus.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {matchRoutes} from 'react-router-config';
+import routesChunkNames from '@generated/routesChunkNames';
+import routes from '@generated/routes'; // eslint-disable-line
+import prefetchHelper from './prefetch';
+import preloadHelper from './preload';
+
+const fetched = {};
+
+const canPrefetchOrLoad = routePath => {
+  // if user is on slow or constrained connection
+  if (`connection` in navigator) {
+    if ((navigator.connection.effectiveType || ``).includes(`2g`)) {
+      return false;
+    }
+    if (navigator.connection.saveData) {
+      return false;
+    }
+  }
+  return !fetched[routePath];
+};
+
+const docusaurus = {
+  prefetch: routePath => {
+    if (!canPrefetchOrLoad(routePath)) {
+      return false;
+    }
+    fetched[routePath] = true;
+
+    // Find all chunk names needed for that particular route
+    const chunkNamesNeeded = [];
+    function dfs(target) {
```

Comment:
Does this make "visited" chunks get added twice? Not too confident of the algorithm here, not very conventional DFS 😝

Reply:
Hahah this algorithm need to be revised. I'll reimplement ha ha ha